x-pack/plugins/apm/common/alert_types.ts

```diff
@@ -7,12 +7,30 @@
 
 import { i18n } from '@kbn/i18n';
 import type { ValuesType } from 'utility-types';
+import * as moment from 'moment';
+import momentDurationFormatSetup from 'moment-duration-format';
 import type { AsDuration, AsPercent } from '../../observability/common';
 import type { ActionGroup } from '../../alerting/common';
 import { ANOMALY_SEVERITY, ANOMALY_THRESHOLD } from './ml_constants';
 
+momentDurationFormatSetup(moment);
```

Comment:
hmm, is this absolutely the only way to do this?

Reply:
What do you mean but _this_? the construction function? or using the Moment Format lib?  @dgieselaar.

Reply:
I'm not a big fan of side effects. Importing this file (or neglecting to do so) could change the behaviour of moment in places we don't expect it. IMHO we should avoid this behaviour. If we _really_ need it, this function should be called in a centralized place. Is it happening anywhere else?

Reply:
I also see this PR adds 12kb to the page load bundle. Presumably that's because of this?

Reply:
That main issue with this file is that the methods that need the dependencies are only used on the server side and the rest of the file is used on the client. This is problematic for 2 reasons:
1. The code shared from the Observability plugin MUST come from `observability/public`
2. Anything you add will increase the bundle size

The solution that Faisal and I discussed was separating those functions out to new file that's only used by the server side. The dependencies will follow and we should end up with a net reduction of the bundle size.

Reply:
Good points! I want to clarify something here. The `format` methods defined in this file are currently used on the client to format the reason message. But as part of this PR @fkanout stopped using this methods on the client  and started reading the reason message from the indexed value. 

At the moment it looks like double work. We call the format methods on the server side to format the message and save it in the alerts-as-data indices and then we call the same method again on the client side to format the reason instead of reading it from the indexed value. Here's the change @fkanout did https://github.com/elastic/kibana/pull/123018/files#diff-13ca656498a368f57a21418f8d444dd369afe2ac0bb7b31b8e42ea3ea5666f84R42 to avoid double formatting on both server side and client side.

Reply:
@mgiota @fkanout is there anything you need from me to solve the bundle size issue? Not sure if you're talking to me or Chris, but want to make sure you're not waiting on anything from me.

Reply:
@dgieselaar Faisal already implemented a simple formatter in this PR https://github.com/elastic/kibana/pull/123794. Once he pulls from upstream, he will get rid of moment and there won't be any bundle size issue. I guess all is needed from you is to approve the PR :-)

Reply:
sure thing, I'll have a look at the bundle size metrics later, assuming the `momentDurationFormatSetup` call then also disappears

Reply:
++ move the 4 functions into the server folder. I would suggest another PR for it.

@simianhacker, It turns out we can expose things that are not in public, in our case `common`, but it should be declared in `kibana.json.` 

@dgieselaar, I moved the duration format function to the o11y plugin folder to be used as a shared resource. And it doesn't use Moment anymore, so should be no impact on the bundle size 😉

Reply:
> The solution that Faisal and I discussed was separating those functions out to new file that's only used by the server side. The dependencies will follow and we should end up with a net reduction of the bundle size.

If I understood correctly, all the format functions that are currently in the [common folder](https://github.com/elastic/kibana/blob/main/x-pack/plugins/apm/common/alert_types.ts) (formatErrorCountReason, formatTransactionErrorRateReason etc) should be moved under `server` folder, since these functions with the current PR are used only on the server side to format the reason before indexing it in the alerts-as-data indices.

On the client side you read the reason from the index https://github.com/elastic/kibana/pull/123018/files#diff-13ca656498a368f57a21418f8d444dd369afe2ac0bb7b31b8e42ea3ea5666f84R42 and you stopped formatting it again. I don't want to block you in this PR, I just think that moving these methods on the server-side would be an improvement.

Reply:
@simianhacker Did I understand your solution correctly. I would move these functions in the server folder anyway, since they are not used by client side anymore

Reply:
@mgiota, Correct, this is what we said with @simianhacker